extensions/amp-analytics/0.1/analytics-platform-vars.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from '../../../src/service';
+import {urlReplacementsFor} from '../../../src/url-replacements';
+import {viewportFor} from '../../../src/viewport';
+
+export class AnalyticsPlatformVars {
+
+  /**
+   * @param {!Window} win
+   * @param {!UrlReplacements} urlReplacements
+   * @param {!Viewport} viewport
+   */
+  constructor(win, urlReplacements, viewport) {
+
+    /** @private {!Map} */
+    this.resolvers_ = new Map();
+
+    /**
+     * @param {!string} name
+     * @param {!Function} resolver
+     */
+    const set = (name, fn) => {this.resolvers_.set(name, fn);};
+
+    /**
+     * @param {string} name
+     * @returns {Function}
+     */
+    const delegateToUrlReplacements = name => {
+      return urlReplacements.get.bind(urlReplacements, name);
+    };
+
+    // Document
+    set('canonicalUrl', delegateToUrlReplacements('CANONICAL_URL'));
```

Comment:
@cramforce wdyt? Are we ok to have two var systems like this? In particular, some of the "new" vars we may also want to have in UrlReplacements as well.